### PR TITLE
Add codeowner workflow check

### PR DESCRIPTION
**Pull request by devindford**
### What does this PR do? What is the motivation?

- Adds a new workflow to ensure that all codeowners for files/directories have to approve the pull request before it can be merged (it will be added as a required job once this is merged)
- Creates/updates a comment about which teams need to approve before a PR can be merged and the status of each codeowner. 
- Adds platform and docs as `CODEOWNER` file reviewers

Passing job after last required approval https://github.com/DataDog/documentation/actions/runs/10292292875/job/28486392695

Failed check on only one approval https://github.com/DataDog/documentation/actions/runs/10292033909
### Merge instructions

- [ ] Please merge after reviewing

### Additional notes

This is an after action item as part of incident web-incident-30 in an effort to ensure the teams that have some type of ownership over files are aware of changes and can review them before they are merged to production

I squashed locally since there was far too many "chore: debug" commits 🙄 so that is why there is a force push

### Note from 08/15/24
The codeowner file was restructured on master and then pulled into this branch, the job outputs listed in the links shows how it worked when there was dual codeowners on the update being made originally, but now only shows the platform approval since we removed Docs from as codeowners on the `CODEOWNERS` file

### CODE REVIEW

**Review by bgdeutsch — APPROVED** [review]

**Review by hestonhoffman — APPROVED** [review]

**Comment by davidejones** [comment]
So discussing this further we determined there are certain cases where a pr will have 2 or more codeowners where we don’t want to enforce all owners to approve. 

This is when a dev team are listed as code owners over parts of the docs but docs team should be able to approve/merge without their consent as they are ultimately the owners of the content.
e.g This pr was just to fix a broken image by removing it https://github.com/DataDog/documentation/pull/24767 which we wouldn't want to block the docs team on etc.

@devindford Can we scope this work to enforce only when our teams are involved?

**Review by hestonhoffman — APPROVED** [review]

**Comment by devindford** [comment]
> So discussing this further we determined there are certain cases where a pr will have 2 or more codeowners where we don’t want to enforce all owners to approve.
> 
> This is when a dev team are listed as code owners over parts of the docs but docs team should be able to approve/merge without their consent as they are ultimately the owners of the content. e.g This pr was just to fix a broken image by removing it #24767 which we wouldn't want to block the docs team on etc.
> 
> @devindford Can we scope this work to enforce only when our teams are involved?

@davidejones After some further discussion with docs, for now they want to trial it the current way and attempt to get all approvals. If this becomes a pain point they will reach out and I will update it to only require Docs (and websites where appropriate) to pass

**Pull request merged**
